src/editor.rs

```diff
@@ -637,13 +644,14 @@ impl Editor {
                 let times = if quit_times > 1 { "times" } else { "time" };
                 set_status!(self, "Press Ctrl+Q {} more {} to quit.", quit_times, times);
             }
-            Key::Char(SAVE) => match self.file_name.take() {
-                // TODO: Can we avoid using take() then reassigning the value to file_name?
-                Some(file_name) => {
-                    self.save_and_handle_io_errors(&file_name);
-                    self.file_name = Some(file_name);
+            Key::Char(SAVE) => {
+                match mem::replace(&mut self.file_name, None) {
```

Comment:
This has the same effect as `take`, I would leave the `take` function in place has it adds more meaning. Your solution also adds an extra line of code.

Reply:
I think you could avoid using `take` by borrowing the value in the `match` arm:

```rust
Key::Char(SAVE) => match self.file_name {
                // TODO: Can we avoid using take() then reassigning the value to file_name?
                Some(ref file_name) => {
                    self.save_and_handle_io_errors(file_name);
                    self.file_name = Some(file_name);
                // ...
```